lxd/auth/driver_tls.go

```diff
@@ -44,12 +45,15 @@ func (t *tls) CheckPermission(ctx context.Context, r *http.Request, object Objec
 		return nil
 	}
 
-	certType, isNotRestricted, projectNames, err := t.certificateDetails(details.username())
+	username := details.username()
+	id, err := t.identities.Get(api.AuthenticationMethodTLS, details.username())
 	if err != nil {
-		return err
+		return fmt.Errorf("Certificate %q missing from identity cache", username)
 	}
 
-	if isNotRestricted || (certType == certificate.TypeMetrics && entitlement == EntitlementCanViewMetrics) {
+	if id.IdentityType == api.IdentityTypeCertificateClientUnrestricted || id.IdentityType == api.IdentityTypeCertificateServer {
```

Comment:
Can we add this logic as a function on `id`, e.g. `id.IsUnrestricted()`? to avoid duplicating the logic here and below?

Reply:
This method actually already exists on the `IdentityType` type in `lxd/db/cluster/identities.go`. Ideally we would use the same type for `IdentityType` everywhere. Would you be opposed to defining it as a type in the `shared/api` package and implementing the `sql.Scanner` and `driver.Valuer` interfaces there?

Reply:
Scrap my last comment. I think I will define some methods in `shared/api/auth.go` that use the string constants. Then we can just use those everywhere and we will only have to make the change in two places (the scanner an valuer methods will still need to be updated but I'd prefer to keep those next to the db logic).

Reply:
Yeah I don't think api should depend on sql